wrappers/src/shared_realm_cs.cpp

```diff
@@ -46,6 +46,39 @@ namespace binding {
     }
 }
     
+    template<typename KeyType>
+    Object* create_object_unique(SharedRealm* realm, Table* table, KeyType key, bool try_update, std::function<size_t (size_t)> find_first, std::function<void (size_t, size_t)> set_unique, std::string key_to_string, bool& is_new)
+    {
+        (*realm)->verify_in_write();
+        const std::string object_name(ObjectStore::object_type_for_table_name(table->get_name()));
```

Comment:
Might as well use Core's `StringData` instead of `std::string` to avoid a heap allocation.